smolyay/test_function_class.py

```diff
@@ -0,0 +1,6478 @@
+import abc
+
+import numpy
+
+class test_fun(abc.ABC):
```

Comment:
Please rename this class either `BenchmarkProblem` or `BenchmarkFunction`. Per PEP8, this is the appropriate naming convention for classes. We are using non-PEP8 names for the implementations below because these are accepted elsewhere.